File: token.py
from token_type import TOKEN_TYPE
from typing import Any


class Token:
    def __init__(self, token_type: TOKEN_TYPE, lexeme: str, literal: Any, line: int):
        self.type: TOKEN_TYPE = token_type
        self.lexeme: str = lexeme
        self.literal: Any = literal
        self.line: int = line

    def __str__(self) -> str:
        return self.type + " " + self.lexeme + " " + self.literal

File: tokenizer/token.go
package tokenizer

import "fmt"

// Token struct
type Token struct {
	typeOfToken TokenType
	lexeme      string
	literal     string
	line        int
}

// NewToken creates new token
func NewToken(typeOfToken TokenType, lexeme string, literal string, line int) *Token {
	token := Token{typeOfToken: typeOfToken, lexeme: lexeme, literal: literal, line: line}
	return &token
}

// ToString string representation of token
func (t *Token) ToString() string {
	return fmt.Sprintf("%d %s %s", t.typeOfToken, t.lexeme, t.literal)
}
